### ISSUE TITLE
vkb::SwapchainBuilder::set_desired_format(VkSurfaceFormatKHR) does not change the format to the requested value.

### ISSUE THREAD

**Issue by PercentBoat4164** [charles-lunarg/vk-bootstrap#48]
Ubuntu 20.04
Vulkan 1.2.172

After running the following in a function that creates my swapchain,
```C++
vkb::SwapchainBuilder swapchainBuilder{ device }; //device is valid and of type vkb::device

VkSurfaceFormatKHR surfaceFormat{};
surfaceFormat.format = VK_FORMAT_R8G8B8A8_SRGB;
surfaceFormat.colorSpace = VK_COLOR_SPACE_SRGB_NONLINEAR_KHR;

//swapchain is of type vkb::Swapchain, and is uninitialized the first time and initialized during all subsequent calls to this function.
vkb::detail::Result<vkb::Swapchain> swap_ret = swapchainBuilder.set_old_swapchain(swapchain).set_desired_format(surfaceFormat).build();
if (!swap_ret) { throw std::runtime_error(swap_ret.error().message()); }
swapchain = swap_ret.value(); //saving the swapchain for to call this function later and for use elsewhere in the program.
```
I attempt to create `VkImage` with a format of `VK_FORMAT_R8G8B8A8_SRGB`; the same format as I used in my swapchain creation, I receive a message from the Vulkan validation layers:
```
[ERROR: Validation]
Validation Error: [ VUID-VkImageViewCreateInfo-image-01019 ] Object 0: handle = 0x70000000007, type = VK_OBJECT_TYPE_IMAGE; | MessageID = 0x3ecd7296 | vkCreateImageView() format VK_FORMAT_R8G8B8A8_SRGB differs from VkImage 0x70000000007[] format VK_FORMAT_B8G8R8A8_UNORM.  Formats MUST be IDENTICAL unless VK_IMAGE_CREATE_MUTABLE_FORMAT BIT was set on image creation. The Vulkan spec states: If image was not created with the VK_IMAGE_CREATE_MUTABLE_FORMAT_BIT flag, format must be identical to the format used to create image (https://vulkan.lunarg.com/doc/view/1.2.170.0~rc2/linux/1.2-extensions/vkspec.html#VUID-VkImageViewCreateInfo-image-01019)
```
As far as I can tell this is saying that I can only create `VkImages` of one `VkFormat`. It does suggest that I use the `VK_IMAGE_CREATE_MUTABLE_FORMAT_BIT` to make this not be an issue, but I thought that the `VkFormat` of the `VkImage` that I tried to make and the `VkFormat` of the swapchain's images were the same. I used a debugger to check the value of `swapchain.image_format` and it was `VK_FORMAT_B8G8R8A8_UNORM` as the validation layer error claimed. It seems to me that this is caused by either my lack of knowledge on how I should be creating my `vkb::Swapchain` or a bug in VkBootstrap.cpp. I looked at your code for setting the format, and it all seemed to make sense. I could not find out how the format ever became `VK_FORMAT_B8G8R8A8_UNORM`. This is very confusing to me as I don't mention that format anywhere in my code, and it is also never mentioned in VkBootstrap.cpp.

If you can make sense of what is going on here any help is welcome. If more information is needed feel free to ask. Thank you.

**Comment by charles-lunarg** [comment]
> I could not find out how the format ever became VK_FORMAT_B8G8R8A8_UNORM
This feels like the requested format is not supported by the hardware and that its using the supported format VK_FORMAT_B8G8R8A8_UNORM. 
But what I am confused about is why you are trying to make images that match the swapchain? The swapchain creates images on your behalf, and you get access to them by asking the API for them (`vkb::Swapchain::get_images()` is the way to do that with vk-bootstrap). 
If this is not the case, then I would ask you to run `vulkaninfo` in the command line or vkconfig (menu option) and find the Surface info. It should list all the supported formats & colorspaces of the hardware.

**Comment by PercentBoat4164** [comment]
Sorry for the confusing way that I wrote things. The images that I was creating were for color and depth attachments as well as one for a texture. Secondly, you were correct. My graphics card and driver combination only supports `FORMAT_B8G8R8A8_UNORM` and `FORMAT_B8G8R8A8_SRGB`. I suppose that this is part of the problem, but I was able to get around it another way. I believe that the error was really caused by my always forcing the `VkFormat` of image views to `VK_FORMAT_R8G8B8A8_SRGB`. I had abstracted `VkImage` creation away, and had not let the `VkImageView`'s format change with the `VkImage`'s. When I fixed this the problem went away.

Thank you for your help. It is good to know about `vulkaninfo`; I will use it in the future. Also, this VkBootstrap package is amazing. Thank you for taking the time to make it. It makes creating a Vulkan engine much easier.
